### PR TITLE
Added AllReduce for IMPI and OMPI

### PR DESCRIPTION
**Pull request by xpillons**
Added AllReduce test with IMPI and OMPI
Added telemetry extraction
Added IMPI options for pingpong test

### CODE REVIEW

File: apps/imb-mpi/ringpingpong.pbs [hunk 1]
```diff
@@ -2,24 +2,31 @@
 
 # setup Intel MPI environment for Infiniband
 source /etc/profile # so we can load modules
-module load mpi/impi_2018.4.274
+module load mpi/impi
```
Comment (edwardsp) [inline apps/imb-mpi/ringpingpong.pbs]: why are we dropping the version here?

File: apps/imb-mpi/ringpingpong.pbs [hunk 2]
```diff
@@ -2,24 +2,31 @@
...
 source $MPI_BIN/mpivars.sh
 
 hostlist=$(pwd)/hosts.$PBS_JOBID
 
 sort -u $PBS_NODEFILE > $hostlist
 
+export I_MPI_FABRICS="shm:dapl"
+export I_MPI_DYNAMIC_CONNECTION=0
```
Comment (edwardsp) [inline apps/imb-mpi/ringpingpong.pbs]: this is only needed for h16r (and due to the lack of support from the microsoft driver)

File: apps/imb-mpi/ringpingpong.pbs [hunk 3]
```diff
@@ -2,24 +2,31 @@
...
 source /etc/profile # so we can load modules
-module load mpi/impi_2018.4.274
+module load mpi/impi
 source $MPI_BIN/mpivars.sh
 
 hostlist=$(pwd)/hosts.$PBS_JOBID
 
 sort -u $PBS_NODEFILE > $hostlist
 
+export I_MPI_FABRICS="shm:dapl"
```
Comment (edwardsp) [inline apps/imb-mpi/ringpingpong.pbs]: this won't work with later versions of intel mpi (and you dropped the version from the module load)

File: apps/imb-mpi/ringpingpong.pbs (hunk 1 above)
Comment (xpillons) [reply apps/imb-mpi/ringpingpong.pbs]: impi is now the generic one for 2018

File: apps/imb-mpi/ringpingpong.pbs (hunk 2 above)
Comment (xpillons) [reply apps/imb-mpi/ringpingpong.pbs]: got it

File: apps/imb-mpi/ringpingpong.pbs (hunk 3 above)
Comment (xpillons) [reply apps/imb-mpi/ringpingpong.pbs]: yes I need to change this based on IMPI version

**Pull request merged**